contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -927,6 +928,248 @@ checksum(PG_FUNCTION_ARGS)
         PG_RETURN_INT32(result);
 }
 
+/*
+ * object_id
+ * 		Returns the object ID with object name and object type as input where object type is option
+ * Returns NULL
+ * 		if input is NULL
+ * 		if there is no such object
+ * 		if user don't have right permission
+ * 		if any error occured
+ */
+Datum
+object_id(PG_FUNCTION_ARGS)
+{	
+	char 			*db_name, *schema_name, *object_name;
+	char 			*physical_schema_name;
+	char 			*input;
+	char 			*object_type = NULL;
+	char 			**splited_object_name;
+	Oid 			schema_oid;
+	Oid				user_id = GetUserId();
+	Oid 			result = InvalidOid;
+	bool 			is_temp_object;
+	int 			i;
+
+	if(PG_ARGISNULL(0))
+		PG_RETURN_NULL();
+	else 
+		input = text_to_cstring(PG_GETARG_TEXT_P(0));
+	if(!PG_ARGISNULL(1))
+		object_type = text_to_cstring(PG_GETARG_TEXT_P(1));
+	
+	/* strip trailing whitespace */
+	i = strlen(input);
+	while (i > 0 && isspace((unsigned char) input[i - 1]))
+		input[--i] = '\0';
+	
+	/* length should be restricted to 4000 */
+	if(strlen(input) > 4000)
+		ereport(ERROR,
+				(errcode(ERRCODE_STRING_DATA_RIGHT_TRUNCATION),
+				errmsg("String or binary data would be truncated.")));
+
+	/* resolve the three part name */
+	splited_object_name = split_object_name(input);
+	db_name = splited_object_name[1];
+	schema_name = splited_object_name[2];
+	object_name = splited_object_name[3];
+
+	/* downcase identifier if needed */
+	if (pltsql_case_insensitive_identifiers)
+	{	
+		db_name = downcase_identifier(db_name, strlen(db_name), false, false);
```

Comment:
are we making a copy or using the same memory?

Reply:
we making copy here, in next commit I have freed the previous memory after use.